cloudflare/resource_cloudflare_argo_tunnel.go

```diff
@@ -0,0 +1,105 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareArgoTunnel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareArgoTunnelCreate,
+		Read:   resourceCloudflareArgoTunnelRead,
+		Update: resourceCloudflareArgoTunnelUpdate,
+		Delete: resourceCloudflareArgoTunnelDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareArgoTunnelImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"secret": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareArgoTunnelCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accID := d.Get("account_id").(string)
+	name := d.Get("name").(string)
+	secret := d.Get("secret").(string)
+
+	tunnel, err := client.CreateArgoTunnel(context.Background(), accID, name, secret)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to create Argo Tunnel"))
+	}
+
+	d.SetId(tunnel.ID)
+
+	return resourceCloudflareArgoTunnelRead(d, meta)
+}
+
+func resourceCloudflareArgoTunnelRead(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceCloudflareArgoTunnelUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
```

Comment:
need to do _something_ here to stop in place updates from being a possibility. will have a dig around and see what options we have.